arbitrator/stylus/src/target_cache.rs

```diff
@@ -56,17 +55,21 @@ pub fn target_cache_set(name: String, description: String, native: bool) -> Resu
             }
             return Err(eyre!(err_message));
         }
-        InitCache::set_target(target.clone())
+        *TARGET_NATIVE.write() = target.clone();
```

Comment:
I'd still add a comment to `target_cache_set` explaining everything this function does, because it kind of does two things in one. It both populates TARGET_CACHE, but if native is set, it also populates TARGET_NATIVE which isn't really a cache or related to TARGET_CACHE.

Reply:
added a following comment: 
```
/// Populates `TARGET_CACHE` inserting target specified by `description` under `name` key.
/// Additionally, if `native` is set it sets `TARGET_NATIVE` to the specified target.
```
Let me know if that's good / precise enough :)